### PR TITLE
Introduce RequestUser decorator

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
public API

### Description
This introduces the `RequestUser` decorator
to extract the `User` from a request.

It reduces code duplication across the public API
and allows us to drop the override of the `Request` type from express.

Signed-off-by: David Mehren <git@herrmehren.de>

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: src/api/public/me/me.controller.ts [hunk 1]
```diff
@@ -65,13 +64,9 @@ export class MeController {
...
   @ApiUnauthorizedResponse({ description: unauthorizedDescription })
-  async getMe(@Req() req: Request): Promise<UserInfoDto> {
-    if (!req.user) {
-      // We should never reach this, as the TokenAuthGuard handles missing user info
-      throw new InternalServerErrorException('Request did not specify user');
-    }
+  async getMe(@RequestUser() user: User): Promise<UserInfoDto> {
     return this.usersService.toUserDto(
-      await this.usersService.getUserByUsername(req.user.userName),
+      await this.usersService.getUserByUsername(user.userName),
```
Comment (InnayTool) [inline src/api/public/me/me.controller.ts:69]:
```suggestion
      user,
```

**Review by InnayTool — CHANGES_REQUESTED** [review]

**Comment by InnayTool** [comment]
I think adding this decorator is pretty nice. Should we do that for `Note` too?

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Comment by DerMolly** [comment]
Should we put the handling of a username in a session in this decorator, too? Or should that be handled in a seperate decoractor? I'm tending towards the second option…

File: src/auth/request-user.decorator.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
```
Comment (DerMolly) [inline src/auth/request-user.decorator.ts:1]: Also, should this file not be in `src/api/utils` like the other decorators?

**Review by DerMolly — APPROVED** [review]

**Pull request merged**